Applications/spire/include/spire/charting/chart_view.hpp

```diff
@@ -65,9 +66,15 @@ namespace Spire {
       ChartValue::Type m_y_axis_type;
       ChartPoint m_top_left;
       ChartPoint m_bottom_right;
+      int m_x_origin;
+      int m_y_origin;
       QFont m_label_font;
       std::string m_timestamp_format;
       CustomVariantItemDelegate* m_item_delegate;
+      QPoint m_crosshair_pos;
+      QImage m_crosshair;
```

Comment:
I suspect the crosshair image is shared among all views in which case it should be defined in an anonymous namespace local to the translation unit using it.

Reply:
I can't put this in an anonymous namespace because the scale() function requires the screen's DPI, which isn't known at compile time. Should I make this a static member then?

Reply:
namespace {
  const auto CROSSHAIR_IMAGE = QImage(scale(x, y));
}

Reply:
It's crashing when I'm running it because there's no qApp instance when this gets called.